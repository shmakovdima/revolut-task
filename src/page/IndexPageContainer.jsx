import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { startCurrencyFetchingAction } from '../store/actions/exchangeRate';
import { startExchangeAction } from '../store/actions/pocket';

import { isDisabled } from '../store/selectors/pocket';

import IndexPageView from './IndexPageView';

const IndexPage = ({ loadCurrency, ...props }) => {
  useEffect(() => {
    loadCurrency();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  return (
    <IndexPageView {...props} />
  );
};

IndexPage.propTypes = {
  loadCurrency: PropTypes.func.isRequired,
  startExchange: PropTypes.func.isRequired,
  disabled: PropTypes.bool.isRequired,
};

const mapStateToProps = state => ({
  disabled: isDisabled(state),
});

const mapDispatchToProps = dispatch => ({
  loadCurrency: () => dispatch(startCurrencyFetchingAction()),
  startExchange: () => dispatch(startExchangeAction()),
});

export { IndexPage };

export default connect(mapStateToProps, mapDispatchToProps)(React.memo(IndexPage));
