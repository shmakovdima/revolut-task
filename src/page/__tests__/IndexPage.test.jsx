import React from 'react';
import { shallow } from 'enzyme';
import { IndexPage } from '../IndexPageContainer';

const props = {
  disabled: false,
  startExchange: jest.fn(),
  loadCurrency: jest.fn(),
};

describe('Component [IndexPage]', () => {
  test('Basic render', () => {
    const component = shallow(<IndexPage
      {...props}
    />);
    expect(component).toMatchSnapshot();
  });
});
