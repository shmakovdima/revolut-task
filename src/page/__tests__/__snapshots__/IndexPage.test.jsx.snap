// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component [IndexPage] Basic render 1`] = `
<Memo(IndexPageView)
  disabled={false}
  startExchange={[MockFunction]}
/>
`;
